lib/file_transfer_agent/gcs_util.js

```diff
@@ -178,7 +178,7 @@ function GCSUtil(httpclient, filestream) {
             encryptionMetadata
           );
         } catch (err) {
-          const errCode = err['code'] ? err['code'] : err.response.status;
+          const errCode = !err['code'] || isNaN(err['code']) ? err.response.status : err['code'];
```

Comment:
Hi @naira-petrosyan-m. Thank you for your proposal. I tried to verify your code and in my opinion it could failed on for example boolean true. What do you think about: 
```!isNaN(err['code']) && !isNaN(parseInt(err['code'])) ? err['code'] : err.response.status ??```

Reply:
@sfc-gh-pmotacki I just thought that err.code wouldn't be a Boolean value in any case, but sure I can change the condition to include parseInt and also parsefloat too.

Reply:
@sfc-gh-pmotacki fyi I have pushed the update above.

Reply:
Thank you @naira-petrosyan-m. The code is correct now. I've run tests on your branch and got failed result. Could you rebase to master branch and verify code using 'npm run lint:check' please.

Reply:
@sfc-gh-pmotacki my branch is opened from the "[gcpoperationerror](https://github.com/snowflakedb/snowflake-connector-nodejs/tree/gcpoperationerror)" branch, and when rebasing with master it gives a lot of conflicts as when rebasing master into "gcpoperationerror". This is what I get when running the lint checking command. 

<img width="602" alt="Screenshot 2024-02-16 at 14 16 52" src="https://github.com/snowflakedb/snowflake-connector-nodejs/assets/143691730/38004a76-d48d-4800-93c8-c91ecb025654">